File: Python Algorithms/binary_search.py
from quick_sort import sort


def binary_search(low, high, key):

    if high >= low:
        mid = int(low + (high - low) / 2)

        if key == array[mid]:
            return mid

        elif key > array[mid]:
            return binary_search(mid, high, key)
        else:
            return binary_search(low, mid, key)
    else:
        return -1


array = [5, 23, 45, 34, 12, 4, 55, 43, 89, 99, 999]
sort(array)  # sort array using quick sort
key = 43
print(binary_search(0, len(array), key))

File: Algorithms/Searching Algorithms/Go/binary_search.go
package main
import "fmt"

// Binary Search Implementation
func binarySearch(needle int, haystack []int) bool {

	low := 0
	high := len(haystack) - 1

	for low <= high{
		median := (low + high) / 2

		if haystack[median] < needle {
			low = median + 1
		}else{
			high = median - 1
		}
	}

	if low == len(haystack) || haystack[low] != needle {
		return false
	}

	return true
}

// Main Function
func main(){
	items := []int{1,2, 9, 20, 31, 45, 63, 70, 100}
	fmt.Println()
	if binarySearch(63, items){
	    fmt.Println("Search is success")
	} else{
	    fmt.Println("Search is unsuccess")
	}
    		
}
